wsmaster/codenvy-hosted-system-ram/src/main/java/com/codenvy/service/systemRam/SystemRamLimitCheckerImpl.java

```diff
@@ -0,0 +1,134 @@
+/*
+ *  [2012] - [2016] Codenvy, S.A.
+ *  All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Codenvy S.A. and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Codenvy S.A.
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.service.systemRam;
+
+import com.google.inject.Inject;
+
+import org.eclipse.che.commons.lang.Size;
+import org.eclipse.che.commons.schedule.ScheduleDelay;
+import org.eclipse.che.plugin.docker.client.DockerConnector;
+import org.eclipse.che.plugin.docker.client.json.SystemInfo;
+import org.everrest.core.impl.provider.json.JsonUtils;
+import org.everrest.websockets.WSConnectionContext;
+import org.everrest.websockets.message.ChannelBroadcastMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Singleton;
+import javax.websocket.EncodeException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeoutException;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Implementation of {@link SystemRamLimitChecker}
+ *
+ * @author Igor Vinokur
+ */
+@Singleton
+public class SystemRamLimitCheckerImpl implements SystemRamLimitChecker{
+
+    private static final Logger LOG = LoggerFactory.getLogger(SystemRamLimitCheckerImpl.class);
+
+    private static final String SYSTEM_RAM_LIMIT_EXCEEDED_STATUS    = "system_ram_limit_exceeded";
+    private static final String SYSTEM_RAM_LIMIT_NOT_EXCEEDED_STATUS = "system_ram_limit_not_exceeded";
+
+    private final DockerConnector dockerConnector;
+    private final ExecutorService executor = Executors.newCachedThreadPool();
+
+    private String limitStatus = SYSTEM_RAM_LIMIT_NOT_EXCEEDED_STATUS;
+    private boolean systemRamLimitExceedMessageSent;
+
+    @Inject
+    public SystemRamLimitCheckerImpl(DockerConnector dockerConnector) {
+        this.dockerConnector = dockerConnector;
+    }
+
+    @Override
+    public final synchronized String getSystemRamLimitStatus() {
+        long systemRamUsed = 0;
+        long systemRamTotal = 0;
+        List<String> allNodesRamUsage = new ArrayList<>();
+        Callable<SystemInfo> task = dockerConnector::getSystemInfo;
+        Future<SystemInfo> future = executor.submit(task);
+        SystemInfo systemInfo;
+        try {
+            systemInfo = future.get(5, SECONDS);
```

Comment:
What is the default timeout of request to docker api? Why doesn't it fit your needs?